### PR TITLE
[stable/envoy] Add lifecyle variable for envoy

### PR DESCRIPTION
**Pull request by renjikari**
Signed-off-by: Shunki Fujiwara <shunki-fujiwara@freee.co.jp>

#### Is this a new chart
No

#### What this PR does / why we need it:
Add lifecyle for Envoy.
It's useful lifecycle events to envoy's rolling update

#### Which issue this PR fixes
  - fixes #

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

### CODE REVIEW

File: stable/envoy/templates/deployment.yaml [hunk 1]
```diff
@@ -88,6 +88,8 @@ spec:
             - name: {{ $key }}
               mountPath: {{ $value.mountPath }}
             {{- end }}
+          lifecycle:
+           {{ toYaml .Values.lifecyle | nindent 12 }}
```
Comment (mumoshu) [inline stable/envoy/templates/deployment.yaml]: I believe you need to add one more space here 😃

**Comment by mumoshu** [comment]
```
Linting chart 'envoy => (version: "1.8.2", path: "stable/envoy")'
Checking chart 'envoy => (version: "1.8.2", path: "stable/envoy")' for a version bump...
Old chart version: 1.8.2
New chart version: 1.8.2
Error linting charts: Error processing charts
------------------------------------------------------------------------------------------------------------------------
 ✖︎ envoy => (version: "1.8.2", path: "stable/envoy") > Chart version not ok. Needs a version bump!
------------------------------------------------------------------------------------------------------------------------
Exited with code 1
```

@renjikari Hey! Could you also bump the minor version number, so that it becomes 1.9.0?

**Comment by renjikari** [comment]
@mumoshu 
Thanks for review!
I fixed your recommendation

File: stable/envoy/templates/deployment.yaml [hunk 2]
```diff
@@ -88,6 +88,8 @@ spec:
             - name: {{ $key }}
               mountPath: {{ $value.mountPath }}
             {{- end }}
+          lifecycle:
+            {{ toYaml .Values.lifecyle | nindent 12 }}
```
Comment (selslack) [inline stable/envoy/templates/deployment.yaml:92]: This doesn't work as `lifecyle` != `lifecycle`. Thanks!

Comment (renjikari) [reply stable/envoy/templates/deployment.yaml:92]: Thanks for report!

I fix this typo at #23258
Please use version after 1.9.1

**Pull request merged**